test-manager/src/tests/mod.rs

```diff
@@ -115,1259 +119,1318 @@ macro_rules! get_possible_api_endpoints {
+        .connect_tunnel(())
+        .await
+        .expect("failed to begin connecting");
+    let new_state = wait_for_tunnel_state(mullvad_client.clone(), |state| {
+        matches!(
+            state,
+            TunnelState::Connecting { .. } | TunnelState::Error(..)
+        )
+    })
+    .await?;
 
-    /// Try to produce leaks in the error state. Refer to the
-    /// `test_connecting_state` documentation for details.
-    #[manager_test]
-    pub async fn test_error_state(
-        rpc: ServiceClient,
-        mut mullvad_client: ManagementServiceClient,
-    ) -> Result<(), Error> {
-        let inet_destination = "1.1.1.1:1337".parse().unwrap();
-        let lan_destination = "172.29.1.200:53".parse().unwrap();
-        let inet_dns = "1.1.1.1:53".parse().unwrap();
-        let lan_dns = "172.29.1.200:53".parse().unwrap();
+    assert!(
+        matches!(new_state, TunnelState::Connecting { .. }),
+        "failed to enter connecting state: {:?}",
+        new_state
+    );
 
-        log::info!("Verify tunnel state: disconnected");
-        assert_tunnel_state!(&mut mullvad_client, TunnelState::Disconnected);
+    //
+    // Leak test
+    //
 
-        //
-        // Connect to non-existent location
-        //
+    assert!(
+        send_guest_probes(rpc.clone(), Some(Interface::NonTunnel), inet_destination)
+            .await?
+            .none(),
+        "observed unexpected outgoing packets (inet)"
+    );
+    assert!(
+        send_guest_probes(rpc.clone(), Some(Interface::NonTunnel), lan_destination)
+            .await?
+            .none(),
+        "observed unexpected outgoing packets (lan)"
+    );
+    assert!(
+        send_guest_probes(rpc.clone(), Some(Interface::NonTunnel), inet_dns)
+            .await?
+            .none(),
+        "observed unexpected outgoing packets (DNS, inet)"
+    );
+    assert!(
+        send_guest_probes(rpc.clone(), Some(Interface::NonTunnel), lan_dns)
+            .await?
+            .none(),
+        "observed unexpected outgoing packets (DNS, lan)"
+    );
 
-        log::info!("Enter error state");
+    assert_tunnel_state!(&mut mullvad_client, TunnelState::Connecting { .. });
 
-        let relay_settings = RelaySettingsUpdate::Normal(RelayConstraintsUpdate {
-            location: Some(Constraint::Only(LocationConstraint::Country(
-                "xx".to_string(),
-            ))),
-            ..Default::default()
-        });
+    //
+    // Disconnect
+    //
 
-        update_relay_settings(&mut mullvad_client, relay_settings)
-            .await
-            .expect("failed to update relay settings");
+    log::info!("Disconnecting");
 
-        let _ = connect_and_wait(&mut mullvad_client).await;
-        assert_tunnel_state!(&mut mullvad_client, TunnelState::Error { .. });
+    disconnect_and_wait(&mut mullvad_client).await?;
 
-        //
-        // Leak test
-        //
+    let relay_settings = RelaySettingsUpdate::Normal(RelayConstraintsUpdate {
```

Comment:
I guess we don't need to manually reset the relay settings here now. Could this be moved to the cleanup function?

Reply:
Yes.